autoload/maktaba/syscall.vim

```diff
@@ -73,12 +73,24 @@ function! s:DoSyscallCommon(syscall, CallFunc, throw_errors) abort
     return l:return_data
   endif
 
+  throw maktaba#syscall#ShellError(a:syscall, l:return_data)
+endfunction
+
+
+""
+" @exception
+" For when a @dict(Syscall) returns an exit code.
+function! maktaba#syscall#ShellError(syscall, return_data) abort
```

Comment:
Note I made this autoload even though it's only used in this file because `@exception` apparently doesn't detect `s:Func`s (see google/vimdoc#1).